test/helpers/xcm.ts

```diff
@@ -875,7 +875,17 @@ export const expectXcmEventMessage = async (context: DevModeContext, message: st
     .map(({ event }) => (context.polkadotJs().events.xcmpQueue.Fail.is(event) ? event : undefined))
     .filter((event) => event);
 
-  return filteredEvents.length ? filteredEvents[0]!.data.error.toString() === message : false;
+  const filtered = filteredEvents[0];
+
+  if (!filtered) {
+    return false;
+  }
+
+  return (
+    context.polkadotJs().events.xcmpQueue.Fail.is(filtered) &&
```

Comment:
This seems redundant.

Reply:
The whole function could be changed to something like:

```ts
export const expectXcmEventMessage = async (context: DevModeContext, message: string) => {
  const records = await context.polkadotJs().query.system.events();

  return records
    .filter(({ event }) => context.polkadotJs().events.xcmpQueue.Fail.is(event))
    .some(({ event }) => event.data.error.toString() === message);
};
```

But I am fine with the current code, was just a small sugestion.

Reply:
done